R/validate-target-data-utils.R

```diff
@@ -0,0 +1,183 @@
+# Target Data Validation Utilities
+
+## OBSERVABLE UNIT VALIDATION ----
+
+#' Validate observable_unit contains only valid columns
+#'
+#' Checks that the observable_unit array only contains:
+#' - Task ID columns (from tasks.json)
+#' - date_col value
+#'
+#' @param config_json Parsed target-data.json as list
+#' @param schema JSON schema for target-data
```

Comment:
I am not sure I understand the differences between these 2 parameters. As I understand, schema is used in the `get_error_path()` function, and in this function schema is defined as

> list representation of a schema object or property containing the property the error is located in.

So, I am not sure what's the difference with `config_json Parsed target-data.json as list`